packages/bpk-component-spinner/src/BpkExtraLargeSpinner.js

```diff
@@ -26,18 +28,26 @@ import SPINNER_TYPES from './spinnerTypes';
 
 const getClassName = cssModules(STYLES);
 
-const BpkExtraLargeSpinner = props => {
+type Props = {
+  type: string,
+  className: ?string,
+};
+
+const BpkExtraLargeSpinner = (props: Props) => {
   const { type, className, ...rest } = props;
   const classNames = [
-    'bpk-spinner',
-    'bpk-spinner--extra-large',
-    `bpk-spinner--${type}`,
-  ].map(getClassName);
+    getClassName(
```

Comment:
Flow didn't like using `.map(getClassName)` as the `getClassName` function returns mixed types depending on what is passed to it, so flow cannot validate it correctly.

However this is better way we get classNames across our components so cleaning it up to align with others

Reply:
This could be simplified a bit more. When we call `getClassName` in this way, it returns a string with the separate classes already separated by spaces. So it feels weird to then wrap the result of that in an array, then do another `.join(' ')` later.

This code will have the same effect in a simpler way:

```jsx
const classNames = getClassName(
  'bpk-spinner',
  'bpk-spinner--extra-large',
  `bpk-spinner--${type}`,
  className
);
```

You can then remove the `if classname` and the `.join(' ')` parts too.

Reply:
Did this suggestion and seems to work fine